enclave/core/sgx/hostcalls.c

```diff
@@ -13,10 +13,14 @@
 static oe_ecall_context_t* _get_ecall_context()
 {
     oe_sgx_td_t* td = oe_sgx_get_td();
-    oe_ecall_context_t* ecall_context = td->host_ecall_context;
-    return oe_is_outside_enclave(ecall_context, sizeof(*ecall_context))
-               ? ecall_context
-               : NULL;
+    if (td)
```

Comment:
There will always be `td`. It is `FS[0]`: 
https://github.com/openenclave/openenclave/blob/5d0ce69f19053a42c5feffec42eb6e3920bf1f00/enclave/core/sgx/td.c#L167-L174

If there is some kind of memory overwrite bug within the enclave, then there is a chance that FS[0] is NULL. In that case, all uses of oe_sgx_get_td ought to be guarded; not just this one.
Our threat model does not consider scenarios where bugs within the enclave cause core data structures to be overwritten. In such scenarios, oe_is_within_enclave check would be safe than a not-null check.